ocaml/sdk-gen/c/autogen/include/xen/api/xen_string_set_set.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) Citrix Systems, Inc.
+ * All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 
+ *   1) Redistributions of source code must retain the above copyright
+ *      notice, this list of conditions and the following disclaimer.
+ * 
+ *   2) Redistributions in binary form must reproduce the above
+ *      copyright notice, this list of conditions and the following
+ *      disclaimer in the documentation and/or other materials
+ *      provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef XEN_STRING_SET_SET_H
+#define XEN_STRING_SET_SET_H
+
+
+#include "xen_common.h"
+#include <xen/api/xen_string_set.h>
+
+
+typedef struct xen_string_set_set
+{
+    size_t size;
+    xen_string_set *contents[];
```

Comment:
Flexible array members without a size need at least C99.
I remember that Visual Studio didn't support C99 (sometimes one had to change the language from C to C++ to get things to compile, but that is not entirely correct either since C++ isn't necessarily a strict superset of C),
other compilers on Windows (and Linux) shouldn't have a problem though.

If you have Visual Studio can you check whether it can compile this as C code? If not a workaround would be to declare the array of having at least 1 member and adjusting the calloc below to allocate n-1 elements (but that is a hack and then I'm not sure it'd be entirely C99 compliant...)

Reply:
It looks like there are now /std:c11 flags since VIsual Studio 2019 version 16.8, so you can try using that (I don't know which version the default implements, their documentation is unclear about that).

Reply:
We always compile it with `-std=c99` (see ocaml/sdk-gen/c/templates/Makefile.mustache).